core/src/nodes/node.rs

```diff
@@ -310,11 +310,18 @@ where
         for (_, outbound) in self.outbound_substreams.drain() {
             self.muxer.destroy_outbound(outbound);
         }
-        if !self.inbound_finished {
-            self.muxer.close_inbound();
-        }
-        if !self.outbound_finished {
-            self.muxer.close_outbound();
+        // TODO: Maybe the shutdown logic should not be part of the destructor?
```

Comment:
The node should definitely get a `shutdown` method that cleanly shuts down the muxer.
This can be in a separate PR though.